CodingTracker.JsPeanut/CodingTracker/CodingController.cs

```diff
@@ -0,0 +1,448 @@
+        {
+            using (var connection = new SqliteConnection(connectionString))
+            {
+                connection.Open();
+
+                var tableCmd = connection.CreateCommand();
+
+                tableCmd.CommandText = "CREATE TABLE IF NOT EXISTS coding_sessions(Id INTEGER PRIMARY KEY AUTOINCREMENT, StartTime TEXT, EndTime TEXT, Duration INTEGER)";
+
+                var tableCmd2 = connection.CreateCommand();
+
+                tableCmd2.CommandText = "CREATE TABLE IF NOT EXISTS goals(Id INTEGER PRIMARY KEY AUTOINCREMENT, Goal TEXT, AddedDate TEXT, ProgressRemaining TEXT)";
+
+                tableCmd.ExecuteNonQuery();
+
+                tableCmd2.ExecuteNonQuery();
+
+                connection.Close();
+            }
+            UserInput.GetUserInput();
+        }
+
+        public static void Insert()
+        {
+            string startTime = UserInput.GetStartTimeInput();
+            DateTime parsedStartTime = DateTime.ParseExact(startTime, "dd/MM/yyyy HH:mm", CultureInfo.CurrentCulture);
+            if (CodingSessions.Any(cs => cs.StartTime == parsedStartTime) == true)
+            {
+                Console.WriteLine("The date and time in which your session started is the same at which other of your sessions started.");
+                Insert();
+            }
+            string endTime = UserInput.GetEndTimeInput();
+            DateTime parsedEndTime = DateTime.ParseExact(endTime, "dd/MM/yyyy HH:mm", CultureInfo.CurrentCulture);
+            if (CodingSessions.Any(cs => cs.EndTime == parsedEndTime) == true)
+            {
+                Console.WriteLine("The date and time in which your session ended is the same at which other of your sessions ended..");
+                Insert();
+            }
+            Validation.ValidateDuration(startTime, endTime, "dd/MM/yyyy HH:mm");
+
+            using (var connection = new SqliteConnection(connectionString))
+            {
+                connection.Open();
+
+                var tableCmd = connection.CreateCommand();
+
+                tableCmd.CommandText = $"INSERT INTO coding_sessions(StartTime, EndTime, Duration) VALUES('{startTime}', '{endTime}', '{DateDifference.Duration}')";
+                tableCmd.ExecuteNonQuery();
+
+                var firstGoal = Goals.FirstOrDefault();
+
+                var tableCmd2 = connection.CreateCommand();
+
+                tableCmd2.CommandText = "SELECT ProgressRemaining FROM goals";
+
+                SqliteDataReader reader = tableCmd2.ExecuteReader();
+
+                TimeSpan progressRemaining = TimeSpan.Zero;
+                if (reader.HasRows)
+                {
+                    while (reader.Read())
+                    {
+                        progressRemaining = TimeSpan.Parse(reader.GetString(0));
+                    }
+                }
+
+                if (DateTime.ParseExact(startTime, "dd/MM/yyyy HH:mm", CultureInfo.CurrentCulture) >= firstGoal.AddedDate)
```

Comment:
I'm getting an error when adding the second date, looks like it's because "firstgoal" is null?  Could you please check? 😊

Reply:
Fixed it. Had to add an if statement which checks if _firstGoal_ is not null, and if it's not then the other if statement will be executed